packages/rest-express/src/transport-express.ts

```diff
@@ -0,0 +1,247 @@
+import AccountsError from '@accounts/error'
+
+import AccountsServer from '@accounts/server';
+
+import { Configuration } from './types/configuration';
+
+import { Router, Request, Response, NextFunction } from 'express';
+
+import { getConnectionInformations } from './utils/get-connection-informations';
+
+interface RequestWithSession extends Request {
+  session: { [key: string]: any };
+}
+
+export default class TransportExpress {
+
+	public router: Router;
+  
+	private accountsServer: AccountsServer;
+	
+	private path: string;
+	
+
+	constructor( config: Configuration ){
+
+		this.path = config.path || 'accounts';
+
+		this.router = Router({ mergeParams: true })
+			.post(`/${this.path}/impersonate`, this.impersonate)
+			.post(`/${this.path}/user`, this.getUser)
+			.post(`/${this.path}/refreshTokens`, this.refreshTokens)
+			.post(`/${this.path}/logout`, this.logout)
+      .post(`/${this.path}/:service/:authenticate`, this.serviceAuthenticate)
+      .post(`/${this.path}/password/register`, this.registerPassword)
+      .post(`/${this.path}/password/verifyEmail`, this.verifyEmail)
+      .post(`/${this.path}/password/resetPassword`, this.resetPassword)
+      .post(`/${this.path}/password/sendVerificationEmail`, this.sendVerificationEmail)
+      .post(`/${this.path}/password/sendResetPasswordEmail`, this.sendResetPasswordEmail)
+      .post(`/${this.path}/password/twoFactorSecret`, this.userLoader, this.twoFactorSecret)
+      .post(`/${this.path}/password/twoFactorSet`, this.userLoader, this.twoFactorSet)
+      .post(`/${this.path}/password/twoFactorUnset`, this.userLoader, this.twoFactorUnset)
+      .post(`/${this.path}/oauth/:provider/callback`, this.providerCallback)
+			// .post(`/${this.path}/:service/:provider?/:action`, this.useService)
+  }
+
+  public link = (accountsServer) => {
```

Comment:
I wanted to keep the Idea of building an AccountsServer from all the parts, and then extract the middleware from the transport but yes, it would achieve the same thing